### PR TITLE
Document errors to ignore instead of forcing it upon users

### PR DESCRIPTION
**Pull request by spawnia**
- [ ] ~Added or updated tests~
- [x] Documented user facing changes
- [x] Updated CHANGELOG.md

Relates to the discussion in https://github.com/nunomaduro/larastan/pull/557

**Changes**

- Document common errors for users to ignore

**Breaking changes**

- Do not overwrite PHPStan's default for `reportUnmatchedIgnoredErrors`
- Stop ignoring errors

### CODE REVIEW

File: extension.neon [hunk 1]
```diff
@@ -26,13 +26,7 @@ parameters:
         - Illuminate\Http\Request
     excludes_analyse:
         - *.blade.php
-    ignoreErrors:
-        - '#Result of function abort \(void\) is used#'
```
Comment (spawnia) [inline extension.neon:30]: Is this a common thing people do? It looks like the result of `abort()` actually **is** `void`.

File: extension.neon [hunk 2]
```diff
@@ -26,13 +26,7 @@ parameters:
...
-        - '#Call to an undefined method Illuminate\\Support\\HigherOrder#'
-        - '#Method App\\Exceptions\\Handler::render\(\) should return Illuminate\\Http\\Response but returns Symfony\\Component\\HttpFoundation\\Response#'
```
Comment (spawnia) [inline extension.neon:32]: That one can be fixed in existing projects and is no longer required in new projects, Laravel cleaned up their act.

File: extension.neon [hunk 3]
```diff
@@ -26,13 +26,7 @@ parameters:
...
-        - '#Property App\\Http\\Middleware\\TrustProxies::\$headers \(string\) does not accept default value of type int#'
```
Comment (spawnia) [inline extension.neon:33]: Also fixable and fixed in the latest Laravel version.

File: extension.neon (hunk 1 above)
Comment (canvural) [reply extension.neon:30]: Yes. I've seen people using `return abort(401)` before. See [here](https://grep.app/search?q=return%20abort%28&filter[lang][0]=PHP) for examples.

Comment (spawnia) [reply extension.neon:30]: It kind of makes the code more clear, i can see why someone would like to do it. Still, i would let that up to personal preference and not mention it in the docs, as it is technically wrong.

**Comment by canvural** [comment]
Should we increase the patch version (0.6) once this is merged?

**Comment by spawnia** [comment]
> Should we increase the minor version (0.6) once this is merged?

Makes sense, there is some breakage that requires a bunch of work. Should i write up a quick upgrade guide?

File: extension.neon (hunk 1 above)
Comment (szepeviktor) [reply extension.neon:30]: Let's tell `return abort`-ers to stop doing it.

**Comment by canvural** [comment]
> Should i write up a quick upgrade guide?

Yes, that would be nice. We can maybe redirect them to the actual [docs](https://github.com/spawnia/larastan/blob/reportUnmatchedIgnoredErrors/docs/errors-to-ignore.md) in the upgrade guide.

File: extension.neon [hunk 4]
```diff
@@ -26,13 +26,7 @@ parameters:
...
     bootstrap: %rootDir%/../../nunomaduro/larastan/bootstrap.php
-    reportUnmatchedIgnoredErrors: false
```
Comment (mfn) [inline extension.neon:35]: Ah, this is great!

I was using phpstan before switching to larastan a few months ago in a private project and, although I knew about this behaviour, I totally forgot about it. When I manually set this flag to true (override) I was prompted with a lot of outdated rules from my codebase! This is so hideous 😄

File: docs/errors-to-ignore.md [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+If you hit those errors in your project, you can add them to your `phpstan.neon` as needed.
+For example:
+
+```neon
+ignoreErrors:
+- '#Result of function abort \(void\) is used#'
+```
+
+> Tip: If you set `reportUnmatchedIgnoredErrors: true`, you will be notified once Larastan
```
Comment (canvural) [inline docs/errors-to-ignore.md:13]: This defaults to `true` in PHPStan. So maybe this tip is not needed.

Comment (spawnia) [reply docs/errors-to-ignore.md]: Replaced with a link to PHPStan's docs.

**Comment by canvural** [comment]
Thank you!

**Pull request merged**